GenderPayGap.WebUI/BusinessLogic/Services/DraftFileBusinessLogic.cs

```diff
@@ -97,11 +98,13 @@ public void KeepDraftFileLockedToUser(Draft draftExpectedToBeLocked, long userId
             SetDraftAccessInformation(userIdRequestingLock, draftExpectedToBeLocked);
         }
 
-        public void DiscardDraft(Draft draftToDiscard)
+        public void DiscardDraft(Draft draftToDiscard, Return submittedReturn)
         {
             List<DraftReturn> matchingReturns = dataRepository.GetAll<DraftReturn>()
                 .Where(dr => dr.OrganisationId == draftToDiscard.OrganisationId)
                 .Where(dr => dr.SnapshotYear == draftToDiscard.SnapshotYear)
+                .AsEnumerable()

```

Comment:
Why are we calling this?

Reply:
EF attempts to evaluate the query on the server where possible. However, on the next line we are calling some external functions that can't be translated into SQL automatically. Therefore, the query needs to be evaluated on the client side here - the data needs to be downloaded into the memory and further parsed (this happens when we call `AsEnumerable()`). Otherwise, an error is thrown.

Reply:
Ah, I see, thanks for the explanation!